lib/SPIRV/libSPIRV/SPIRVFunction.cpp

```diff
@@ -150,7 +150,10 @@ SPIRVFunction::decodeBB(SPIRVDecoder &Decoder) {
 
     SPIRVInstruction *Inst = static_cast<SPIRVInstruction *>(Decoder.getEntry());
     assert(Inst);
-    BB->addInstruction(Inst);
+    if (Inst->getOpCode() == OpUndef)
+      Module->addUndef(Inst->getType());
```

Comment:
I think, we just don't need to call addUndef, because Decoder.getEntry() crates an entry and adds it to the module. Besides, in this way, id of the original Undef instructions is kept.